File: 2012m./uzd2./sprendimai/uzd2.py
class God:
    def __init__(self, name, score):
        self.name = name
        self.score = score

def readData():
    data = open("U2.txt", "r")

    basicData = next(data).split(" ")

    numOfGods = int(basicData[0])
    numOfDice = int(basicData[1])

    gods = []
    for i in range(numOfGods):
        godData = next(data)
        name = godData[:10]
        score = 0
        godData = godData[10:].split(" ")
        for j in range(numOfDice):
            score += int(godData[j])
        gods.append(God(name, score))

    data.close()

    return gods

def findLord(gods):
    lordIndex = 0
    for i in range(1, len(gods)):
        if(gods[i].score > gods[lordIndex].score):
            lordIndex = i
    return gods[lordIndex]

def main():
    output = open("U2rez.txt", "w")

    gods = readData()
    lord = findLord(gods)

    output.write(f"{lord.name} {lord.score}")

    output.close()

if __name__ == "__main__":
    main()

File: 2012m./uzd2./sprendimai/uzd2.cpp
#include <iostream>
#include <fstream>
#include <sstream>
#include <vector>

using namespace std;

struct God {
    string name;
    int score;

    God(const string name, int score) : name(name), score(score) {}
};

vector<God> readData() {
    ifstream data("U2.txt");

    vector<God> gods;

    int numOfGods, numOfDice;
    data >> numOfGods >> numOfDice;
    data.ignore();

    for (int i = 0; i < numOfGods; i++) {
        string godData;
        getline(data, godData);
        string name = godData.substr(0, 10);

        stringstream datastream(godData.substr(10)); 

        int score = 0;
        for (int j = 0; j < numOfDice; j++) {
            int diceValue;
            datastream >> diceValue;
            score += diceValue;
        }

        gods.push_back(God(name, score));
    }

    data.close();

    return gods;
}

God findLord(const vector<God> gods) {
    int lordIndex = 0;
    for (size_t i = 1; i < gods.size(); i++) {
        if (gods[i].score > gods[lordIndex].score) {
            lordIndex = static_cast<int>(i);
        }
    }
    return gods[lordIndex];
}

int main() {
    ofstream output("U2rez.txt");

    vector<God> gods = readData();
    God lord = findLord(gods);

    output << lord.name << " " << lord.score;

    output.close();

    return 0;
}